swarm/network/stream/snapshot_retrieval_test.go

```diff
@@ -321,11 +326,14 @@ func runRetrievalTest(chunkCount int, nodeCount int) error {
 						log.Debug(fmt.Sprintf("File with root hash %x successfully retrieved", hash))
 					}
 				}
-				allSuccess = localSuccess
+				if !localSuccess {
+					allSuccess = false
+					break
+				}
 			}
 		}
 		if !allSuccess {
-			return fmt.Errorf("Not all chunks succeeded!")
+			return fmt.Errorf("Not all retrievals succeeded!")
```

Comment:
this is unreachable code right? you never break from the outer loop above as long as `!allSuccess`

Reply:
right, `break` on line 331 doesn't break from the outer loop